src/modules/dashboard/components/CreateEditDomainDialog/CreateEditDomainDialogForm.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { useState } from 'react';
+import { ColonyRole } from '@colony/colony-js';
+import { FormikProps } from 'formik';
+import { FormattedMessage, defineMessages } from 'react-intl';
+
+import Button from '~core/Button';
+import ColorSelect from '~core/ColorSelect';
+import { Color } from '~core/ColorTag';
+import DialogSection from '~core/Dialog/DialogSection';
+import { Input, Annotations } from '~core/Fields';
+import Heading from '~core/Heading';
+import PermissionsLabel from '~core/PermissionsLabel';
+import PermissionRequiredInfo from '~core/PermissionRequiredInfo';
+
+import { Colony, useLoggedInUser } from '~data/index';
+import { useTransformer } from '~utils/hooks';
+
+import { getAllUserRoles } from '../../../transformers';
+import { canArchitect } from '../../../users/checks';
+
+import { FormValues } from './CreateEditDomainDialog';
+import styles from './CreateEditDomainDialogForm.css';
+
+const MSG = defineMessages({
+  titleCreate: {
+    id:
+      'dashboard.CreateEditDomainDialog.CreateEditDomainDialogForm.titleCreate',
+    defaultMessage: 'Create a new domain',
+  },
+  titleEdit: {
+    id: 'dashboard.CreateEditDomainDialog.CreateEditDomainDialogForm.titleEdit',
+    defaultMessage: 'Edit domain details',
+  },
+  name: {
+    id: 'dashboard.CreateEditDomainDialog.CreateEditDomainDialogForm.name',
+    defaultMessage: 'Domain name',
+  },
+  purpose: {
+    id: 'dashboard.CreateEditDomainDialog.CreateEditDomainDialogForm.name',
+    defaultMessage: 'What is the purpose of this domain?',
+  },
+  annotation: {
+    id:
+      'dashboard.CreateEditDomainDialog.CreateEditDomainDialogForm.annotation',
+    defaultMessage: 'Explain why you’re creating this domain',
+  },
+  noPermissionFrom: {
+    id:
+      'dashboard.CreatePaymentDialog.CreatePaymentDialogForm.noPermissionFrom',
+    defaultMessage:
+      // eslint-disable-next-line max-len
+      'You need the {roleRequired} permission in {domain} to take this action.',
+  },
+});
+
+interface Props {
+  back: () => void;
+  colony: Colony;
+  id?: string;
+}
+
+const CreateEditDomainDialogForm = ({
+  back,
+  colony,
+  handleSubmit,
+  id,
+}: Props & FormikProps<FormValues>) => {
+  const [domainColor, setDomainColor] = useState(Color.LightPink);
+
+  const { walletAddress, username, ethereal } = useLoggedInUser();
+
+  const allUserRoles = useTransformer(getAllUserRoles, [colony, walletAddress]);
+
+  const hasRegisteredProfile = !!username && !ethereal;
+  const canCreateEditDomain =
+    hasRegisteredProfile && canArchitect(allUserRoles);
+
+  return (
+    <>
+      <DialogSection>
+        <Heading
+          appearance={{ size: 'medium', margin: 'none' }}
+          text={id === undefined ? MSG.titleCreate : MSG.titleEdit}
+        />
```

Comment:
The styles for this heading are different from the specs / other modals.

Yours:
![Screenshot from 2021-01-05 14-43-37](https://user-images.githubusercontent.com/1193222/103649758-af86b980-4f67-11eb-966d-2afb5f7b8330.png)

Another modal:
![Screenshot from 2021-01-05 14-43-44](https://user-images.githubusercontent.com/1193222/103649760-b0b7e680-4f67-11eb-8679-8582bfbe5923.png)

Please change this so the title styles are aligned